packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -191,34 +208,38 @@ class _BottomAppBarState extends State<BottomAppBar> {
     final double? height = widget.height ?? babTheme.height ?? defaults.height;
     final Color color = widget.color ?? babTheme.color ?? defaults.color!;
     final Color surfaceTintColor = widget.surfaceTintColor ?? babTheme.surfaceTintColor ?? defaults.surfaceTintColor!;
-    final Color effectiveColor = isMaterial3 ? color : ElevationOverlay.applyOverlay(context, color, elevation);
     final Color shadowColor = widget.shadowColor ?? babTheme.shadowColor ?? defaults.shadowColor!;
 
-    final Widget child = Padding(
+    Widget child = Padding(
       padding: widget.padding ?? babTheme.padding ?? (isMaterial3 ? const EdgeInsets.symmetric(vertical: 12.0, horizontal: 16.0) : EdgeInsets.zero),
       child: widget.child,
     );
 
-    final Material material = Material(
-      key: materialKey,
-      type: isMaterial3 ? MaterialType.canvas : MaterialType.transparency,
-      elevation: elevation,
-      color: isMaterial3 ? effectiveColor : null,
-      surfaceTintColor: surfaceTintColor,
-      shadowColor: shadowColor,
-      child: SafeArea(child: child),
-    );
-
-    final PhysicalShape physicalShape = PhysicalShape(
-      clipper: clipper,
-      elevation: elevation,
-      shadowColor: shadowColor,
-      color: effectiveColor,
-      clipBehavior: widget.clipBehavior,
-      child: material,
-     );
+    if (isMaterial3) {
+      child = Material(
+        key: materialKey,
+        elevation: elevation,
+        color: color,
+        surfaceTintColor: surfaceTintColor,
+        shadowColor: shadowColor,
+        child: SafeArea(child: child),
+      );
+    } else {
+      child = PhysicalShape(
+        clipper: clipper,
+        elevation: elevation,
+        shadowColor: shadowColor,
+        color: ElevationOverlay.applyOverlay(context, color, elevation),
+        clipBehavior: widget.clipBehavior,
+        child: Material(
+          key: materialKey,
+          type: MaterialType.transparency,
+          child: SafeArea(child: child),
```

Comment:
This is exactly like how this code was before Material 3 support was first added in #106525, except for the customizations added for padding in #115175 and for shadow color in #121406.